components/docked-composer/index.jsx

```diff
@@ -0,0 +1,166 @@
+	/**
+	 * Body of the composer.
+	 * _Tested with snapshot testing._
+	 */
+	body: PropTypes.node,
+	/**
+	 * **Assistive text for accessibility**
+	 * This object is merged with the default props object on every render.
+	 * * `expandButton`: This is a visually hidden label for expand button shown when the composer is minimized.
+	 * * `minimizeButton`: This is a visually hidden label for minimize button shown when the composer is expanded.
+	 * * `closeButton`: This is a visually hidden label for close button.
+	 * _Tested with snapshot testing._
+	 */
+	assistiveText: PropTypes.shape({
+		expandButton: PropTypes.string,
+		minimizeButton: PropTypes.string,
+		closeButton: PropTypes.string,
+	}),
+	/**
+	 * Additional class name added to the container with slds-docked_container class.
+	 * _Tested with snapshot testing._
+	 */
+	classNameContainer: PropTypes.string,
+	/**
+	 * Event Callbacks
+	 * * `onMinimize`: Called when minimize button is clicked.
+	 * * `onExpand`: Called when the expand button is clicked.
+	 * * `onClose`: Called when the close button is clicked.
+	 * _Tested with Mocha testing._
+	 */
+	events: PropTypes.shape({
+		onMinimize: PropTypes.func,
+		onExpand: PropTypes.func,
+		onClose: PropTypes.func,
+	}),
+};
+
+/**
+ * The Docked Composer is a persistent utility bar that allows a user to continually use the app to complete tasks or gather information while expanding/collapsing a composer window. [component blueprint guidelines](https://www.lightningdesignsystem.com/components/docked-composer/).
+ */
+class DockedComposer extends React.Component {
+	constructor(props) {
+		super(props);
+		this.generatedId = shortid.generate();
+	}
+
+	getId() {
+		return this.props.id || this.generatedId;
+	}
+
+	render() {
+		const sectionClassName = this.props.isOpen
+			? 'slds-is-open'
+			: 'slds-is-closed';
+		return (
+			<div
+				id={`${this.getId()}-container`}
+				className={`slds-docked_container${
+					this.props.classNameContainer
+						? ` ${this.props.classNameContainer}`
+						: ''
+				}`}
+			>
+				<section
+					className={`slds-docked-composer slds-grid slds-grid_vertical ${sectionClassName}`}
+					role="dialog"
+					aria-labelledby="dialog-heading-id-1"
+					aria-describedby="dialog-content-id-1"
+				>
+					<header
+						className="slds-docked-composer__header slds-grid slds-shrink-none"
+						aria-live="assertive"
+					>
+						<div className="slds-media slds-media_center slds-no-space">
+							<div className="slds-media__body">
+								<h2
+									className="slds-truncate"
+									id="dialog-heading-id-1"
+									title={this.props.header}
+								>
+									{this.props.header}
+								</h2>
+							</div>
+						</div>
+						<div className="slds-col_bump-left slds-shrink-none">
+							{this.props.isOpen ? (
+								<Button
+									id={`${this.getId()}-minimize-button`}
+									title={this.props.assistiveText?.minimizeButton || 'Minimize'}
```

Comment:
I think was would be typically done by a default props object, so no one has to search render functions for text literals.